File: samplejava/src/main/java/pyxis/uzuki/live/richutils/samplejava/ContactActivity.java
package pyxis.uzuki.live.richutils.samplejava;

import android.Manifest;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.TextView;

import java.util.List;

import butterknife.BindView;
import pyxis.uzuki.live.richutilskt.utils.ContactItem;
import pyxis.uzuki.live.richutilskt.utils.RPermission;
import pyxis.uzuki.live.richutilskt.utils.RichUtils;

public class ContactActivity extends BaseActivity {
    @BindView(R.id.txtResult) TextView txtResult;
    String text = "";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_contacts);

        String[] arrays = new String[]{Manifest.permission.READ_CONTACTS};

        RPermission.getInstance(this).checkPermission(arrays, (integer, strings) -> {
            init();
        });
    }

    private void init() {
        RichUtils.runAsync(() -> {
            List<ContactItem> list = RichUtils.getContactsList(ContactActivity.this);
            for (ContactItem item : list) {
                text += "\n" + item;
            }

            runOnUiThread(() -> txtResult.setText(text));
        });
    }
}

File: sample/src/main/java/pyxis/uzuki/live/richutilssample/ContactActivity.kt
package pyxis.uzuki.live.richutilssample

import android.Manifest
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_contacts.*
import pyxis.uzuki.live.richutilskt.utils.*

class ContactActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_contacts)

        val arrays: Array<String> = arrayOf(Manifest.permission.READ_CONTACTS)
        val isGranted = RPermission.getInstance(this).checkPermission(array = arrays, callback = { _: Int, _: ArrayList<String> ->
            init()
        })

        if (isGranted)
            init()
    }

    private fun init() {
        var text = ""
        val progress = progress(message = "Loading...")
        runAsync {
            val list = getContactsList()
            list.forEach { text = "$text \n$it" }

            runOnUiThread {
                progress.dismiss()
                txtResult.text = text
            }
        }
    }
}